src/PdfViewer/index.tsx

```diff
@@ -0,0 +1,93 @@
+import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
+import 'react-pdf/dist/esm/Page/TextLayer.css';
+
+import { FileEntry } from "@tauri-apps/api/fs";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { Document, Page } from "react-pdf";
+
+import { readFile } from "../filesystem";
+import { PdfViewerAPI } from '../types/PdfViewerAPI';
+
+export interface PdfViewerProps {
+  file: FileEntry;
+  pdfViewerRef: React.MutableRefObject<PdfViewerAPI | null>;
+}
+
+export const PdfViewer = ({ file, pdfViewerRef }: PdfViewerProps) => {
+  const [numPages, setNumPages] = useState<number | null>(null);
+
+  const [fileContent, setFileContent] = useState<{ data: Uint8Array } | null>(null);
+  const [isFileLoading, setIsFileLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [pdfViewerWidth, setPdfViewerWidth] = useState<number>();
+
+  const updateTimeoutRef = useRef<NodeJS.Timeout>();
+
+  const updateWidth = useCallback(() => {
+    clearTimeout(updateTimeoutRef.current);
+    updateTimeoutRef.current = setTimeout(() => {
+      setPdfViewerWidth(containerRef.current?.getBoundingClientRect().width);
+    }, 200);
+  }, [updateTimeoutRef]);
+
+  // Update viewer's width on window resize
+  useEffect(() => {
```

Comment:
Can't we use the https://usehooks-ts.com/react-hook/use-element-size instead of this effect?